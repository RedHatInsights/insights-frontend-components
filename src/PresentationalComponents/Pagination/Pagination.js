import React, { Component } from 'react';
import { PaginationRow } from 'patternfly-react';
import PropTypes from 'prop-types';
import debounce from 'lodash/debounce';

import { generateID } from '../../functions/generateID.js';

export const dropDirection = {
    up: 'up',
    down: 'down'
};

const pager = [ 10, 15, 20, 25, 50 ];

class Pagination extends Component {
    constructor(props) {
        super(props);
        this.defaultFirstPage = this.defaultFirstPage.bind(this);
        this.defaultLastPage = this.defaultLastPage.bind(this);
        this.defaultPreviousPage = this.defaultPreviousPage.bind(this);
        this.defaultNextPage = this.defaultNextPage.bind(this);
        this.setPage = this.setPage.bind(this);
    }

    setPage(page) {
        const perPage = this.props.itemsPerPage || pager[0];
        const maxPage = Math.ceil(this.props.numberOfItems / perPage);
        page = isNaN(page) ? 1 : page;
        page = page > maxPage ? maxPage : page;
        this.props.hasOwnProperty('onSetPage') && this.props.onSetPage(page);
    }

    defaultFirstPage() {
        this.setPage(1);
    }

    defaultLastPage() {
        const perPage = this.props.itemsPerPage || pager[0];
        this.setPage(Math.ceil(this.props.numberOfItems / perPage));
    }

    defaultPreviousPage() {
        const { page = 1 } = this.props;
        this.setPage(page - 1);
    }

    defaultNextPage() {
        const { page = 1 } = this.props;
        this.setPage(page + 1);
    }

    render() {
        const { page = 1 } = this.props;
        const perPage = this.props.itemsPerPage || pager[0];
        const perPageOptions = this.props.perPageOptions || pager;
        const lastPage = Math.ceil(this.props.numberOfItems / perPage);
        const lastIndex = page === lastPage ? this.props.numberOfItems : page * perPage;
        const firstIndex = page === 1 ? 1 : page * perPage - perPage;
        return (
            <div className="special-patternfly" widget-type='InsightsPagination'>
                <PaginationRow
                    { ...this.props }
                    pageInputValue={ this.props.page || 1 }
                    viewType={ this.props.viewType || 'table' }
                    pagination={ { perPage, page, perPageOptions } }
                    amountOfPages={ lastPage }
                    itemCount={ this.props.numberOfItems }
                    itemsStart={ firstIndex }
                    pageSizeDropUp={ this.props.direction === 'up' }
                    itemsEnd={ lastIndex }
                    onPerPageSelect={ this.props.onPerPageSelect }
                    onPageInput={ event => debounce(this.setPage(parseInt(event.target.value), 10), 100) }
                    onFirstPage={ this.props.onFirstPage || this.defaultFirstPage }
                    onLastPage={ this.props.onLastPage || this.defaultLastPage }
                    onPreviousPage={ this.props.onPreviousPage || this.defaultPreviousPage }
                    onNextPage={ this.props.onNextPage || this.defaultNextPage }
                />
            </div>
        );
    }
}

Pagination.propTypes = {
    direction: PropTypes.oneOf(Object.keys(dropDirection)),
    viewType: PropTypes.string,
    itemsPerPage: PropTypes.number,
    perPageOptions: PropTypes.arrayOf(PropTypes.number),
    numberOfItems: PropTypes.number.isRequired,
    onSetPage: PropTypes.func,
    onPerPageSelect: PropTypes.func
};

export default Pagination;
