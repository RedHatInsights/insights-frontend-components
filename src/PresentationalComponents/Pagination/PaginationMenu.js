import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CaretDownIcon, CaretUpIcon } from '@patternfly/react-icons';
import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core';

class PaginationNav extends Component {
    state = {
        isOpen: false
    }

    onSelect = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    };

    onToggle = isOpen => {
        this.setState({
            isOpen
        });
    };

    render() {
        const {
            itemsTitle,
            itemsStart,
            itemsEnd,
            widtgetId,
            dropDirection,
            onSetPerPage,
            itemCount,
            perPageOptions,
            className,
            ...props
        } = this.props;
        const { isOpen } = this.state;
        return (
            <div className={ `pf-c-options-menu ${className}` } { ...props }>
                <span id={ `${widtgetId}-label` } hidden>Items per page:</span>
                <div className="pf-c-options-menu__toggle pf-m-text pf-m-plain">
                    <Dropdown direction={ dropDirection }
                        isPlain
                        isOpen={ isOpen }
                        onSelect={ this.onSelect }
                        dropdownItems={ perPageOptions.map(({ title, value }) => (
                            <DropdownItem onClick={ event => onSetPerPage(event, value) } key={ value } component="button">{ title }</DropdownItem>
                        )) }
                        toggle={
                            <DropdownToggle onToggle={ this.onToggle } iconComponent={ null } className="pf-c-options-menu__toggle-button">
                                <span className="pf-c-options-menu__toggle-text">
                                    <b>{ itemsStart } - { itemsEnd }</b> of <b>{ itemCount }</b> { itemsTitle }
                                </span>
                                { dropDirection === 'up' ? <CaretUpIcon /> : <CaretDownIcon /> }
                            </DropdownToggle>
                        }
                    />
                </div>
            </div>
        );
    }
}

PaginationNav.propTypes = {
    itemsTitle: PropTypes.string,
    itemsStart: PropTypes.number,
    itemsEnd: PropTypes.number,
    dropDirection: PropTypes.string,
    widtgetId: PropTypes.string,
    onSetPerPage: PropTypes.func,
    itemCount: PropTypes.number.isRequired,
    perPageOptions: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.node,
        value: PropTypes.number
    }))
};

PaginationNav.defaultProps = {
    itemsTitle: 'items',
    dropDirection: 'up'
};

export default PaginationNav;
