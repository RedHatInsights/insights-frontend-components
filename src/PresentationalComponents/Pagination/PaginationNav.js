import React from 'react';
import PropTypes from 'prop-types';
import { Button, ButtonVariant } from '@patternfly/react-core';
import { AngleDoubleLeftIcon, AngleLeftIcon, AngleDoubleRightIcon, AngleRightIcon } from '@patternfly/react-icons';

const PaginationNav = ({ lastPage, setPage, pageTitle, amountOfPages, page, ...props }) => {
    return <nav className="pf-c-pagination__nav" aria-label="Pagination" { ...props }>
        <Button variant={ ButtonVariant.plain }
            isDisabled={ page === 1 }
            aria-label="Go to first page"
            onClick={ event => setPage(event, 1) }
        >
            <AngleDoubleLeftIcon />
        </Button>
        <Button variant={ ButtonVariant.plain }
            isDisabled={ page === 1 }
            aria-label="Go to previous page"
            onClick={ event => setPage(event, page - 1) }
        >
            <AngleLeftIcon />
        </Button>
        <div className="pf-c-pagination__nav-page-select" aria-label="Current page 1 of 4">
            <input className="pf-c-form-control"
                aria-label="Current page"
                type="number"
                min="1"
                max={ amountOfPages }
                size="2"
                value={ page }
                onChange={ event => setPage(event, Number(event.target.value)) }
            />
            <span aria-hidden="true">of { lastPage } { pageTitle }</span>
        </div>
        <Button variant={ ButtonVariant.plain }
            isDisabled={ page === lastPage }
            aria-label="Go to next page"
            onClick={ event => setPage(event, page + 1) }
        >
            <AngleRightIcon />
        </Button>
        <Button variant={ ButtonVariant.plain }
            isDisabled={ page === lastPage }
            aria-label="Go to last page"
            onClick={ event => setPage(event, lastPage) }
        >
            <AngleDoubleRightIcon />
        </Button>
    </nav>;
};

PaginationNav.propTypes = {
    lastPage: PropTypes.number,
    page: PropTypes.number,
    pageTitle: PropTypes.string,
    setPage: PropTypes.func,
    amountOfPages: PropTypes.number
};

PaginationNav.defaultProps = {
    pageTitle: 'pages'
};

export default PaginationNav;
