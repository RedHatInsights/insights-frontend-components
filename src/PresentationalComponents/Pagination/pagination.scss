$fa-font-path:   "~font-awesome-sass/assets/fonts/font-awesome/";
@import "~font-awesome-sass/assets/stylesheets/font-awesome";

.special-patternfly {
   $font-path: '~patternfly/dist/fonts/';
   $img-path: '~patternfly/dist/img/';
   $icon-font-path: '~patternfly/dist/fonts/';
   @import '~patternfly/dist/sass/_patternfly.scss';
   @import '~patternfly-react/dist/sass/_patternfly-react.scss';
}
