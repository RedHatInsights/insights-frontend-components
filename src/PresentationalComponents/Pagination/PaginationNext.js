import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PaginationNav from './PaginationNav';
import PaginationMenu from './PaginationMenu';
import '@patternfly/patternfly-next/components/Pagination/pagination.scss';
import '@patternfly/patternfly-next/components/OptionsMenu/options-menu.scss';

class PaginationNext extends Component {
    render() {
        const {
            className,
            page,
            lastPage,
            setPage,
            itemsStart,
            itemsEnd,
            widtgetId,
            onSetPerPage,
            itemCount,
            perPageOptions,
            dropDirection,
            amountOfPages,
            ...props
        } = this.props;
        return (
            <div className={ `pf-c-pagination pf-m-footer ${className}` }  aria-label="Element pagination" { ...props }>
                <PaginationMenu
                    itemsStart={ itemsStart }
                    itemsEnd={ itemsEnd }
                    widtgetId={ widtgetId }
                    dropDirection={ dropDirection }
                    onSetPerPage={ onSetPerPage }
                    itemCount={ itemCount }
                    perPageOptions={ perPageOptions }
                />
                <PaginationNav lastPage={ lastPage } page={ page } setPage={ setPage } />
            </div>
        );
    }
}

PaginationNext.propTypes = {
    amountOfPages: PropTypes.number,
    itemCount: PropTypes.number.isRequired,
    page: PropTypes.number,
    itemsStart: PropTypes.number,
    itemsEnd: PropTypes.number,
    widtgetId: PropTypes.string,
    dropDirection: PropTypes.string,
    onSetPerPage: PropTypes.func,
    perPageOptions: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.node,
        value: PropTypes.number
    })),
    lastPage: PropTypes.number,
    setPage: PropTypes.func
};

export default PaginationNext;
