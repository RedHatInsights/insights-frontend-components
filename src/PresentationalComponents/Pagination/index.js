export { default as Pagination } from './pagination';
import './pagination.scss';