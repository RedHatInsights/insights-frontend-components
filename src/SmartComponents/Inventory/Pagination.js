import React, { Component, Fragment } from 'react';
import { Pagination, dropDirection } from '../../PresentationalComponents/Pagination';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { InventoryContext } from './Inventory';
import debounce from 'lodash/debounce';

const perPageOptions = [ 10, 20, 50, 100 ];

class ContextFooterPagination extends Component {
    constructor(props) {
        super(props);
        this.state = {
            page: undefined
        };

        this.changePage = debounce((pagination) => this.props.onRefreshData(pagination), 800);
    }

    onSetPage = (page, debounce) => {
        const { perPage, filters, onRefreshData } = this.props;
        // eslint-disable-next-line camelcase
        const pagination = { page, per_page: perPage, filters };
        if (debounce) {
            this.changePage(pagination);
            this.setState({
                page
            });
        } else {
            onRefreshData(pagination);
            this.setState({
                page: undefined
            });
        }
    }

    onPerPageSelect = (perPage) => {
        const { page, filters, onRefreshData } = this.props;
        // eslint-disable-next-line camelcase
        onRefreshData({ page, per_page: perPage, filters });
    }

    render() {
        const { total, page, perPage, loaded } = this.props;
        const { page: statePage } = this.state;
        return (
            <Fragment>
                { loaded && (
                    <Pagination
                        numberOfItems={ total }
                        perPageOptions={ perPageOptions }
                        page={ statePage || page }
                        itemsPerPage={ perPage }
                        direction={ dropDirection.up }
                        onSetPage={ this.onSetPage }
                        onPerPageSelect={ this.onPerPageSelect }
                    />
                ) }
            </Fragment>
        );
    }
}

const FooterPagination = ({ ...props }) => (
    <InventoryContext.Consumer>
        { ({ onRefreshData }) => (
            <ContextFooterPagination { ...props } onRefreshData={ onRefreshData } />
        ) }
    </InventoryContext.Consumer>
);

const propTypes = {
    perPage: PropTypes.number,
    total: PropTypes.number,
    loaded: PropTypes.bool
};

ContextFooterPagination.propTypes = {
    ...propTypes,
    onRefreshData: PropTypes.func
};

FooterPagination.propTypes = propTypes;

FooterPagination.defaultProps = {
    total: 0,
    loaded: false,
    onRefreshData: () => undefined
};

function stateToProps({ entities: { page, perPage, total, loaded, activeFilters }}, { totalItems }) {
    return {
        page,
        perPage,
        total: totalItems || total,
        loaded,
        filters: activeFilters
    };
}

export default connect(stateToProps, null)(FooterPagination);
