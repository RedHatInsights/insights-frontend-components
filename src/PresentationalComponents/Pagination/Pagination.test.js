import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination, { dropDirection } from './Pagination';

describe('Pagination component', () => {
    describe('should render', () => {
        it('required props', () => {
            const wrapper = shallow(<Pagination numberOfItems={10} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('per page options', () => {
            const wrapper = shallow(<Pagination numberOfItems={10} perPageOptions={[1,2]}/>);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('items per page', () => {
            const wrapper = shallow(<Pagination numberOfItems={10} itemsPerPage={20} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('direction up', () => {
            const wrapper = shallow(<Pagination numberOfItems={10} direction={dropDirection.up} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('empty page', () => {
            const wrapper = shallow(<Pagination numberOfItems={0} />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        describe('page', () => {
            it('second', () => {
                const wrapper = shallow(<Pagination numberOfItems={30} page={2} />);
                expect(toJson(wrapper)).toMatchSnapshot();
            });

            it('last', () => {
                const wrapper = shallow(<Pagination numberOfItems={30} page={3} />);
                expect(toJson(wrapper)).toMatchSnapshot();
            });
        });
    });
    describe('API', () => {
        describe('onSetPage', () => {
            it('first page', () => {
                const onSetPage = jest.fn();
                const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                wrapper.find('.pagination-pf-back a[title="First Page"]').first().simulate('click');
                expect(onSetPage.mock.calls[0][0]).toBe(1);
            });

            it('last page', () => {
                const onSetPage = jest.fn();
                const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                wrapper.find('.pagination-pf-forward a[title="Last Page"]').first().simulate('click');
                expect(onSetPage.mock.calls[0][0]).toBe(3);
            });

            it('previous page', () => {
                const onSetPage = jest.fn();
                const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                wrapper.find('.pagination-pf-back a[title="Previous Page"]').first().simulate('click');
                expect(onSetPage.mock.calls[0][0]).toBe(1);
            });

            it('next page', () => {
                const onSetPage = jest.fn();
                const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                wrapper.find('.pagination-pf-forward a[title="Next Page"]').first().simulate('click');
                expect(onSetPage.mock.calls[0][0]).toBe(3);
            });

            describe('input', () => {
                it('number', () => {
                    const onSetPage = jest.fn();
                    const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                    wrapper.find('input.pagination-pf-page').first().simulate('change', { target: { value: '1' } });
                    expect(onSetPage.mock.calls[0][0]).toBe(1);
                });

                it('large', () => {
                    const onSetPage = jest.fn();
                    const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                    wrapper.find('input.pagination-pf-page').first().simulate('change', { target: { value: '150' } });
                    expect(onSetPage.mock.calls[0][0]).toBe(3);
                });

                it('string', () => {
                    const onSetPage = jest.fn();
                    const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                    wrapper.find('input.pagination-pf-page').first().simulate('change', { target: { value: 'bad' } });
                    expect(onSetPage.mock.calls[0][0]).toBe(2);
                });

                it('negative page', () => {
                    const onSetPage = jest.fn();
                    const wrapper = mount(<Pagination numberOfItems={30} onSetPage={onSetPage} page={2} />);
                    wrapper.find('input.pagination-pf-page').first().simulate('change', { target: { value: '-1' } });
                    expect(onSetPage.mock.calls[0][0]).toBe(0);
                });
            });
        });

        it('onPerPageSelect', () => {
            const onPerPageSelect = jest.fn();
            const wrapper = mount(<Pagination numberOfItems={30} onPerPageSelect={onPerPageSelect} />);
            wrapper.find('[aria-labelledby="pagination-row-dropdown"] li a').at(2).simulate('click');
            expect(onPerPageSelect.mock.calls.length).toBe(1);
            expect(onPerPageSelect.mock.calls[0][0]).toBe(20);
        });

        it('onFirstPage', () => {
            const onFirstPage = jest.fn();
            const wrapper = mount(<Pagination numberOfItems={30} onFirstPage={onFirstPage} page={2}/>);
            wrapper.find('.pagination-pf-back a[title="First Page"]').first().simulate('click');
            expect(onFirstPage.mock.calls.length).toBe(1);
        });

        it('onLastPage', () => {
            const onLastPage = jest.fn();
            const wrapper = mount(<Pagination numberOfItems={30} onLastPage={onLastPage} />);
            wrapper.find('.pagination-pf-forward a[title="Last Page"]').first().simulate('click');
            expect(onLastPage.mock.calls.length).toBe(1);
        });

        it('onPreviousPage', () => {
            const onPreviousPage = jest.fn();
            const wrapper = mount(<Pagination numberOfItems={30} onPreviousPage={onPreviousPage} page={2} />);
            wrapper.find('.pagination-pf-back a[title="Previous Page"]').first().simulate('click');
            expect(onPreviousPage.mock.calls.length).toBe(1);
        });

        it('onNextPage', () => {
            const onNextPage = jest.fn();
            const wrapper = mount(<Pagination numberOfItems={30} onNextPage={onNextPage} page={2} />);
            wrapper.find('.pagination-pf-forward a[title="Next Page"]').first().simulate('click');
            expect(onNextPage.mock.calls.length).toBe(1);
        });
    });
});
