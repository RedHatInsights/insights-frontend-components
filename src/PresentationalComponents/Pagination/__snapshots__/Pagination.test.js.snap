// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render direction up 1`] = `
<div
  className="special-patternfly"
  widget-type="InsightsPagination"
>
  <PaginationRow
    amountOfPages={1}
    baseClassName="content-view-pf-pagination"
    className=""
    direction="up"
    dropdownButtonId="pagination-row-dropdown"
    itemCount={10}
    itemsEnd={10}
    itemsStart={1}
    messages={
      Object {
        "currentPage": "Current Page",
        "firstPage": "First Page",
        "lastPage": "Last Page",
        "nextPage": "Next Page",
        "of": "of",
        "perPage": "per page",
        "previousPage": "Previous Page",
      }
    }
    numberOfItems={10}
    onFirstPage={[Function]}
    onLastPage={[Function]}
    onNextPage={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousPage={[Function]}
    onSubmit={[Function]}
    page={1}
    pageInputValue={1}
    pageSizeDropUp={true}
    pagination={
      Object {
        "page": 1,
        "perPage": 10,
        "perPageOptions": Array [
          10,
          15,
          20,
          25,
          50,
        ],
      }
    }
    viewType="table"
  />
</div>
`;

exports[`Pagination component should render empty page 1`] = `
<div
  className="special-patternfly"
  widget-type="InsightsPagination"
>
  <PaginationRow
    amountOfPages={1}
    baseClassName="content-view-pf-pagination"
    className=""
    dropdownButtonId="pagination-row-dropdown"
    itemCount={0}
    itemsEnd={0}
    itemsStart={0}
    messages={
      Object {
        "currentPage": "Current Page",
        "firstPage": "First Page",
        "lastPage": "Last Page",
        "nextPage": "Next Page",
        "of": "of",
        "perPage": "per page",
        "previousPage": "Previous Page",
      }
    }
    numberOfItems={0}
    onFirstPage={[Function]}
    onLastPage={[Function]}
    onNextPage={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousPage={[Function]}
    onSubmit={[Function]}
    page={1}
    pageInputValue={1}
    pageSizeDropUp={false}
    pagination={
      Object {
        "page": 1,
        "perPage": 10,
        "perPageOptions": Array [
          10,
          15,
          20,
          25,
          50,
        ],
      }
    }
    viewType="table"
  />
</div>
`;

exports[`Pagination component should render items per page 1`] = `
<div
  className="special-patternfly"
  widget-type="InsightsPagination"
>
  <PaginationRow
    amountOfPages={1}
    baseClassName="content-view-pf-pagination"
    className=""
    dropdownButtonId="pagination-row-dropdown"
    itemCount={10}
    itemsEnd={10}
    itemsPerPage={20}
    itemsStart={1}
    messages={
      Object {
        "currentPage": "Current Page",
        "firstPage": "First Page",
        "lastPage": "Last Page",
        "nextPage": "Next Page",
        "of": "of",
        "perPage": "per page",
        "previousPage": "Previous Page",
      }
    }
    numberOfItems={10}
    onFirstPage={[Function]}
    onLastPage={[Function]}
    onNextPage={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousPage={[Function]}
    onSubmit={[Function]}
    page={1}
    pageInputValue={1}
    pageSizeDropUp={false}
    pagination={
      Object {
        "page": 1,
        "perPage": 20,
        "perPageOptions": Array [
          10,
          15,
          20,
          25,
          50,
        ],
      }
    }
    viewType="table"
  />
</div>
`;

exports[`Pagination component should render page last 1`] = `
<div
  className="special-patternfly"
  widget-type="InsightsPagination"
>
  <PaginationRow
    amountOfPages={3}
    baseClassName="content-view-pf-pagination"
    className=""
    dropdownButtonId="pagination-row-dropdown"
    itemCount={30}
    itemsEnd={30}
    itemsStart={20}
    messages={
      Object {
        "currentPage": "Current Page",
        "firstPage": "First Page",
        "lastPage": "Last Page",
        "nextPage": "Next Page",
        "of": "of",
        "perPage": "per page",
        "previousPage": "Previous Page",
      }
    }
    numberOfItems={30}
    onFirstPage={[Function]}
    onLastPage={[Function]}
    onNextPage={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousPage={[Function]}
    onSubmit={[Function]}
    page={3}
    pageInputValue={3}
    pageSizeDropUp={false}
    pagination={
      Object {
        "page": 3,
        "perPage": 10,
        "perPageOptions": Array [
          10,
          15,
          20,
          25,
          50,
        ],
      }
    }
    viewType="table"
  />
</div>
`;

exports[`Pagination component should render page second 1`] = `
<div
  className="special-patternfly"
  widget-type="InsightsPagination"
>
  <PaginationRow
    amountOfPages={3}
    baseClassName="content-view-pf-pagination"
    className=""
    dropdownButtonId="pagination-row-dropdown"
    itemCount={30}
    itemsEnd={20}
    itemsStart={10}
    messages={
      Object {
        "currentPage": "Current Page",
        "firstPage": "First Page",
        "lastPage": "Last Page",
        "nextPage": "Next Page",
        "of": "of",
        "perPage": "per page",
        "previousPage": "Previous Page",
      }
    }
    numberOfItems={30}
    onFirstPage={[Function]}
    onLastPage={[Function]}
    onNextPage={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousPage={[Function]}
    onSubmit={[Function]}
    page={2}
    pageInputValue={2}
    pageSizeDropUp={false}
    pagination={
      Object {
        "page": 2,
        "perPage": 10,
        "perPageOptions": Array [
          10,
          15,
          20,
          25,
          50,
        ],
      }
    }
    viewType="table"
  />
</div>
`;

exports[`Pagination component should render per page options 1`] = `
<div
  className="special-patternfly"
  widget-type="InsightsPagination"
>
  <PaginationRow
    amountOfPages={10}
    baseClassName="content-view-pf-pagination"
    className=""
    dropdownButtonId="pagination-row-dropdown"
    itemCount={10}
    itemsEnd={1}
    itemsStart={1}
    messages={
      Object {
        "currentPage": "Current Page",
        "firstPage": "First Page",
        "lastPage": "Last Page",
        "nextPage": "Next Page",
        "of": "of",
        "perPage": "per page",
        "previousPage": "Previous Page",
      }
    }
    numberOfItems={10}
    onFirstPage={[Function]}
    onLastPage={[Function]}
    onNextPage={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousPage={[Function]}
    onSubmit={[Function]}
    page={1}
    pageInputValue={1}
    pageSizeDropUp={false}
    pagination={
      Object {
        "page": 1,
        "perPage": 1,
        "perPageOptions": Array [
          1,
          2,
        ],
      }
    }
    perPageOptions={
      Array [
        1,
        2,
      ]
    }
    viewType="table"
  />
</div>
`;

exports[`Pagination component should render required props 1`] = `
<div
  className="special-patternfly"
  widget-type="InsightsPagination"
>
  <PaginationRow
    amountOfPages={1}
    baseClassName="content-view-pf-pagination"
    className=""
    dropdownButtonId="pagination-row-dropdown"
    itemCount={10}
    itemsEnd={10}
    itemsStart={1}
    messages={
      Object {
        "currentPage": "Current Page",
        "firstPage": "First Page",
        "lastPage": "Last Page",
        "nextPage": "Next Page",
        "of": "of",
        "perPage": "per page",
        "previousPage": "Previous Page",
      }
    }
    numberOfItems={10}
    onFirstPage={[Function]}
    onLastPage={[Function]}
    onNextPage={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousPage={[Function]}
    onSubmit={[Function]}
    page={1}
    pageInputValue={1}
    pageSizeDropUp={false}
    pagination={
      Object {
        "page": 1,
        "perPage": 10,
        "perPageOptions": Array [
          10,
          15,
          20,
          25,
          50,
        ],
      }
    }
    viewType="table"
  />
</div>
`;
